src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,410 @@
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature);
+        }
+
+        private readonly Bindings _bindings;
+
+        /// <summary>
+        /// Tensorflow session object
+        /// </summary>
+        private readonly TFSession _session;
+
+        /// <summary>
+        /// Tensorflow saves model with 'serve' tag when it is intended for serving.
+        /// </summary>
+        private const string ServingTag = "serve";
+
+        public override ISchema Schema => _bindings;
+
+        /// <summary>
+        ///  First dimension in each tensor is a batch dimension.
+        ///  Currently setting it to 1.
+        /// </summary>
+        private const int BatchSize = 1;
+
+        /// <summary>
+        /// Convenience constructor for public facing API.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
+        /// <param name="modelDir">Directory where the tensorflow model is saved.</param>
+        /// <param name="name">Name of the output column. Keep it same as in the Tensorflow model.</param>
+        /// <param name="source">Name of the input column(s). Keep it same as in the Tensorflow model.</param>
+        public TensorflowTransform(IHostEnvironment env, IDataView input, string modelDir, string name, params string[] source)
+            : this(env, new Arguments() { Column = new[] { new Column() { Source = source, Name = name } }, ModelDir = modelDir }, input)
+        {
+        }
+
+        public TensorflowTransform(IHostEnvironment env, Arguments args, IDataView input)
+            : base(env, RegistrationName, input)
+        {
+            Host.CheckValue(args, nameof(args));
+            Host.CheckUserArg(Utils.Size(args.Column) > 0, nameof(args.Column));
+            for (int i = 0; i < args.Column.Length; i++)
+                Host.CheckUserArg(Utils.Size(args.Column[i].Source) > 0, nameof(args.Column));
+
+            _session = LoadTFSession(args.ModelDir);
+            _bindings = new Bindings(args.Column, Source.Schema, this);
+        }
+
+        private TFSession LoadTFSession(string modelDir)
+        {
+            var sessionOptions = new TFSessionOptions();
+            var runOptions = new TFBuffer();
+            var graph = new TFGraph();
+            var metaGraphDef = new TFBuffer();
+            var status = new TFStatus();
+
+            return new TFSession().FromSavedModel(
+                    sessionOptions,
+                    runOptions,
+                    modelDir,
+                    new[] { ServingTag },
+                    graph,
+                    metaGraphDef);
+        }
+
+        public override void Save(ModelSaveContext ctx)
+        {
+            throw new NotImplementedException();
```

Comment:
>throw [](start = 12, length = 5)

Have you given some thought on how we are planning to save/load such transforms?

Reply:
I will look into it. I haven't thought about it yet. Maybe I will look into CNTK implementation in TLC to see what they are doing.

---
In reply to: [209306191](https://github.com/zeahmed/machinelearning/pull/3#discussion_r209306191) [](ancestors = 209306191)